radixdlt-java/src/main/java/com/radixdlt/client/application/translate/AtomErrorToExceptionReasonMapper.java

```diff
@@ -0,0 +1,9 @@
+package com.radixdlt.client.application.translate;
+
+import com.google.gson.JsonObject;
+import com.radixdlt.client.core.atoms.Atom;
+import java.util.stream.Stream;
+
+public interface AtomErrorToExceptionReasonMapper {
+	Stream<ActionExecutionExceptionReason> mapAtomErrorToExceptionReasons(Atom atom, JsonObject errorData);
```

Comment:
Why is this a `Stream`? Not an `Observable`, `List` or whatever.

Reply:
Not an `Observable` since it's not asynchronous. Not a `List` as the result is mainly aggregated in which a `Stream` is much more efficient.